edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs

```diff
@@ -62,7 +62,8 @@ public async Task TestEdgeHubConnection()
             string sasKey = ConnectionStringHelper.GetSharedAccessKey(deviceConnStr);
             var signatureProvider = new SharedAccessKeySignatureProvider(sasKey);
             var credentialsCache = Mock.Of<ICredentialsCache>();
-            var metadataStore = Mock.Of<IMetadataStore>();
+            var metadataStore = new Mock<IMetadataStore>();
+            metadataStore.Setup(m => m.GetMetadata(It.IsAny<string>())).ReturnsAsync(new ConnectionMetadata("dummyValue"));
```

Comment:
Why is this change needed?

Reply:
The metadata store object needs to be mocked because there is a GetMetadata in the new flow.
This GetMetadata is safe and will always return a ConnectionMetadata object, but if we don't mock it, the GetMetadata call fails and the test breaks

Reply:
The metadata store object needs to be mocked because there is a MetadataStore.GetMetadata in the new flow.
This GetMetadata is safe and will always return a ConnectionMetadata object, but if we don't mock it, the GetMetadata call fails and the test breaks